test/zauction.test.ts

```diff
@@ -344,4 +325,49 @@ describe("zAuction Contract Tests", () => {
 
     await expect(tx).to.be.revertedWith("zAuction: data already consumed");
   });
+  it("Calculates minter royalty correctly", async () => {
+    // A percent with 5 decimals of precision
+    mockRegistrar.domainRoyaltyAmount.returns(1000000);
+    // Each WILD is 10^18, Bid is 15 WILD
+    const bid = ethers.utils.parseEther("15");
+    const id = "12345";
+    const royalty = await zAuction.calculateMinterRoyalty(bid, id);
+    const decimal = royalty.toString();
+
+    // 10% of bid
+    expect(decimal).to.equal(ethers.utils.parseEther("1.5"));
+  });
+  it("Calculates root owner royalty correctly", async () => {
+    // Each WILD is 10^18, Bid is 15 WILD
+    // A percent with 5 decimals of precision
+    const bid = ethers.utils.parseEther("15");
+    const id = "123245";
+    mockRegistrar.domainRoyaltyAmount.returns(1000000);
+    const callers = await ethers.getSigners();
+    mockRegistrar.ownerOf.whenCalledWith(id).returns(callers[0].address);
+
+    await zAuction.connect(callers[0]).setRootRoyaltyAmount(id, 10);
+
+    const royalty = await zAuction.calculateRootOwnerRoyalty(id, bid, id);
+    const decimal = royalty.toString();
+
+    // 10% of bid (one less zero)
+    expect(decimal).to.equal(ethers.utils.parseEther("1.5"));
+  });
+  it("Gets the root parent of a domain that is already the root", async () => {
+    // Case where id given is already the root
+    mockRegistrar.parentOf.returns(0);
+    const id = "12345";
+    const rootId = await zAuction.rootDomainIdOf(id);
+    expect(rootId).to.equal(id);
+  });
+  it("Gets the root parent when the id given is not already the root", async () => {
+    // Case where id given is not the root
+    mockRegistrar.parentOf.whenCalledWith("3").returns("2");
+    mockRegistrar.parentOf.whenCalledWith("2").returns("1");
+    mockRegistrar.parentOf.whenCalledWith("1").returns("0");
+    const id = "3";
+    const rootId = await zAuction.rootDomainIdOf(id);
+    expect(rootId).to.equal("1");
+  });
 });
```

Comment:
Consider writing a test scenario with these cases:

`Domain ABC` is being sold by `User X` to `User Y`
`User X` placed a bid of `123` tokens (thus the sale is 123 tokens)
`Domain ABC` has a royalty of `5%` which is owed to `User Z`
`Domain ABC`'s top level domain is `Domain A` which has a `4.44%` fee which is owned by `User W`

This test will ensure that
`mockERC20Token.transferFrom` is called 3 times with the following parameters
User X -> User Z for 6.150 tokens
User X -> User W for 5.461 tokens
User X -> User Y for 111.389 tokens
(Granted there will be some decimal cut off, so you'll have to do some rounding or truncation to the nearest 1000th decimal place)